hallpass/passes/views.py

```diff
@@ -15,36 +15,46 @@ def home(request):
 def monitor_destinations(request):
     user_profile = request.user.profile
     user_destinations = user_profile.destinations.filter(building = user_profile.building)
-    form = CreateHallPassForm(request = request)
+    # form = CreateHallPassForm(request = request)
+    form = CreateHallPassForm(request.POST, request = request)
 
     if not user_destinations:
         return redirect(reverse('dashboard'))
 
     hallpasses = HallPass.objects.filter(Time_out = None)
+    count = len(HallPass.objects.exclude(Time_in = None).filter(Time_out = None))
     if request.method == 'POST':
-        if 'Enter' in request.POST['action']:
-            form = CreateHallPassForm(request.POST, request = request)
+        if 'Enter' in request.POST['action']: 
             if form.is_valid(): 
-                student_id = form.cleaned_data['student']
-                student = Student.objects.filter(student_id=student_id)[0]
                 d = Destination.objects.get(id = request.POST['action'].split(" ")[1])
-
-                HallPass.objects.filter(student_id = student).update(Time_out = datetime.datetime.now())
+                student_id = form.cleaned_data['student']   
+                student = Student.objects.filter(student_id=student_id)[0]
+           
+                def queueCheck():
+                    if len(HallPass.objects.filter(student_id = student)) >= 1:
+                        HallPass.objects.filter(student_id = student).update(Time_out = datetime.datetime.now())
+                    count = len(HallPass.objects.exclude(Time_in = None).filter(Time_out = None))
+                    if d.max_people_allowed > count :
+                        return datetime.datetime.now()
```

Comment:
I don't like this. Our function only returns if condition is true. Otherwise it just implicitly returns none. This doesn't seem like a great practice. When you call this it's returning a now() or None. How are you handling this in the event of None?
https://stackoverflow.com/a/15300671/7799574